### ISSUE TITLE
ios: Background mode querying results in Realm error (v0.12.0)

### ISSUE THREAD

**Issue by willks** [realm/realm-js#424]
Hi,
  I'm using RN 0.22.2, and stumbled upon what I believe is an issue. I created an event which fires every minute (using GCD) which then periodically executes a JS function which in turn, accesses RealmDB for data to sync up with server. When the application is put in the background, Realm fails with this error when it is accessed:

`2016-05-06 14:42:18.729 [fatal][tid:com.facebook.React.RCTExceptionsManagerQueue] Unhandled JS Exception: Failed to execute 'send' on 'XMLHttpRequest': Failed to load 'http://localhost:8082/call_method'.`

More details:

`sendRequest - rpc.js@170`
`Object.callMethod - rpc.js @82:0`
`Realm.<anonymous> - util.js @ 65:0`
`Realm.objects @ 106:0 `

I had this implementation today with Async store, and that worked perfectly - but I was unhappy with Async as a database and only Realm made any real sense long term.

<img width="373" alt="error" src="https://cloud.githubusercontent.com/assets/8967148/15064742/30d74706-139b-11e6-919d-c69668661f65.png">

TLDR stuff - ignore if you will:

I see that Realm does support background tasks in iOS/Android based on the searches I've done, specifically: https://realm.io/docs/swift/latest/, which uses GCD. I'm using GCD only as an even trigger every minute to execute an event like this:

https://gist.github.com/willks/c1b41a03906b756b0981c664b191dd84

**Comment by appden** [comment]
@willks That error indicates that the app is running in Chrome debug mode. The way Chrome debug mode works with Realm is that it runs an HTTP server embedded in the app that the Realm JS needs to make requests to. When the app is backgrounded, it probably stops responding to those requests. When you take the app out of Chrome debugging mode, does everything work as expected?

**Comment by willks** [comment]
Yes! After I typed this up, I actually deployed to phone and sue enough, all the data was uploaded while app was in background on my phone. I'm new to RN - and I appreciate your quick response.

Verified as working and down to my silly testing.

Thanks!

**Comment by frangeris** [comment]
Hey @appden is this equivalent or the same?

<img width="1420" alt="screen shot 2016-07-21 at 8 20 56 pm" src="https://cloud.githubusercontent.com/assets/811381/17042022/b378adfa-4f80-11e6-803f-5c0c527e7459.png">

**Comment by alazier** [comment]
@frangeris this looks like a different error. Looks like you are running on device and connections from chrome running on your computer to the device are failing. Are you seeing the same error in simulator?
